#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_correct_errorcode_on_failure.dpatch by Stefan Fritsch <sf@sfritsch.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: don't return 0 on error

@DPATCH@
diff -urNad mp3gain-1.5.1~/mp3gain.c mp3gain-1.5.1/mp3gain.c
--- mp3gain-1.5.1~/mp3gain.c	2009-07-24 13:57:12.000000000 +0200
+++ mp3gain-1.5.1/mp3gain.c	2009-07-24 15:23:32.065420323 +0200
@@ -1985,6 +1985,7 @@
 					} else {
 						fprintf(stderr,"No undo information in %s\n",argv[mainloop]);
 					}
+					gSuccess = 0;
 				}
 		  }
 	  }
@@ -2051,8 +2052,9 @@
 #else
 		  if ((inf == NULL)&&(tagInfo[mainloop].recalc > 0)) {
 #endif
-			  fprintf(stdout, "Can't open %s for reading\n",argv[mainloop]);
-              fflush(stdout);
+			  fprintf(stderr, "Can't open %s for reading\n",argv[mainloop]);
+			  fflush(stderr);
+			  gSuccess = 0;
 		  }
 		  else {
 #ifdef AACGAIN
@@ -2137,8 +2139,9 @@
 				if (!ok) {
 #endif
                     if (!BadLayer) {
-						fprintf(stdout,"Can't find any valid MP3 frames in file %s\n",argv[mainloop]);
-                        fflush(stdout);
+						fprintf(stderr,"Can't find any valid MP3 frames in file %s\n",argv[mainloop]);
+						fflush(stderr);
+						gSuccess = 0;
                     }
 				}
 				else {
@@ -2173,8 +2176,8 @@
 								(Xingcheck[0] == 'I' && Xingcheck[1] == 'n' && Xingcheck[2] == 'f' && Xingcheck[3] == 'o')) {
 							bitridx = (curframe[2] >> 4) & 0x0F;
 							if (bitridx == 0) {
-								fprintf(stdout, "%s is free format (not currently supported)\n",curfilename);
-								fflush(stdout);
+								fprintf(stderr, "%s is free format (not currently supported)\n",curfilename);
+								fflush(stderr);
 								ok = 0;
 							}
 							else {
@@ -2217,8 +2220,8 @@
 						while (ok) {
 							bitridx = (curframe[2] >> 4) & 0x0F;
 							if (bitridx == 0) {
-								fprintf(stdout,"%s is free format (not currently supported)\n",curfilename);
-								fflush(stdout);
+								fprintf(stderr,"%s is free format (not currently supported)\n",curfilename);
+								fflush(stderr);
 								ok = 0;
 							}
 							else {
@@ -2304,8 +2307,9 @@
 					}
 
 					if (dBchange == GAIN_NOT_ENOUGH_SAMPLES) {
-						fprintf(stdout,"Not enough samples in %s to do analysis\n",argv[mainloop]);
-                        fflush(stdout);
+						fprintf(stderr,"Not enough samples in %s to do analysis\n",argv[mainloop]);
+                        			fflush(stderr);
+						gSuccess = 0;
 						numFiles--;
 					}
 					else {
@@ -2445,8 +2449,8 @@
 		}
 
 		if (dBchange == GAIN_NOT_ENOUGH_SAMPLES) {
-			fprintf(stdout,"Not enough samples in mp3 files to do analysis\n");
-            fflush(stdout);
+			fprintf(stderr,"Not enough samples in mp3 files to do analysis\n");
+			fflush(stderr);
 		}
 		else {
 			Float_t maxmaxsample;
